src/actions/CryptoExchangeActions.js

```diff
@@ -120,28 +125,24 @@ export const setNativeAmount = (info: SetNativeAmountInfo) => (dispatch: any, ge
 
   // make spend
   if (fromWallet && toWallet) {
-    dispatch(getShiftTransaction(fromWallet, toWallet)).catch((e) => {
-      console.log(' ERROR getting shidt transaction. ')
+    try {
+      await dispatch(getShiftTransaction(fromWallet, toWallet))
+    } catch (e) {
       console.log(e)
       if (e.name === Constants.INSUFFICIENT_FUNDS || e.message === Constants.INSUFFICIENT_FUNDS) {
         dispatch(actions.dispatchAction(Constants.RECEIVED_INSUFFICIENT_FUNDS_ERROR))
         return
       }
-      if (e.message === Constants.DUST) {
-        dispatch(actions.dispatchAction(Constants.RECEIVED_DUST_ERROR))
-        return
-      }
       dispatch(actions.dispatchActionString(Constants.GENERIC_SHAPE_SHIFT_ERROR, e.message))
-    })
+    }
   }
 }
 
-export const shiftCryptoCurrency = () => async  (dispatch: any, getState: any) => {
+export const shiftCryptoCurrency = () => async  (dispatch: Dispatch, getState: GetState) => {
   const state = getState()
   const srcWallet: AbcCurrencyWallet = CORE_SELECTORS.getWallet(state, state.cryptoExchange.fromWallet.id)
-  if (!srcWallet) {
-    return
-  }
+
+  if (!srcWallet) { return }
```

Comment:
Please don't do these types of cleanup. These are harder to set breakpoints on and just make the code review more polluted.